sampler.go

```diff
@@ -3,32 +3,53 @@ package otelutil
 import (
 	"go.opentelemetry.io/otel/attribute"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
 )
 
-type AlwaysErrorSampler struct {
-	base sdktrace.Sampler
+// sampler -  if the span has the specified attributes and the base sampler would have sampled it, then it will be sampled.
+type attributeSampler struct {
+	base  sdktrace.Sampler
+	attrs map[attribute.Key]attribute.Value
```

Comment:
So once one of the attributes exists with the same value, then we Record the sample? Basically this is an OR operation, if at least one key/value matches we are recording the span. Is that right?

Reply:
yes, the idea behind it is that when we create the span using `tracer.Start(..)` the sampler will kick in and look for the attrs to see if it needs to sample